src/lib.rs

```diff
@@ -193,8 +276,8 @@ impl MediaType {
     /// Implements the [MIME Sniffing standard]
     /// (https://mimesniff.spec.whatwg.org/#mime-type-groups) for MIME type groups.
     pub fn is_zip_based_type(&self) -> bool {
-        u(&self.suffix) == Some("zip")
-        || MediaType::new(APPLICATION, STANDARDS, Some("zip"), None).eq_mime_portion(self)
+        self.suffix() == Some(&"zip".into()) ||
```

Comment:
If suffix etc returns `&str` then a lot of these test comparisons should simplify (although you'll probably get type errors anyway to guide you)